src/core/dbus-manager.c

```diff
@@ -2255,7 +2255,7 @@ static int method_link_unit_files(sd_bus_message *message, void *userdata, sd_bu
 }
```

Comment:
hmm, so instead of propagating down the full preset mode, even in contexts where we aren't running a preset operation, what about just adding a new flag to UnitFileFlags, maybe called UNIT_FILE_RELAX_INTERFACE_PROPAGATION or something like that? THen, set thtat flag only in the preset mode. Benefit: no need to add one more param to functions that already have way too many params, and no passing of a param that conceptually doesn't make sense in most execution contexts.